buildtools/pipenv/pipenv.go

```diff
@@ -0,0 +1,148 @@
+package pipenv
+
+import (
+	"encoding/json"
+
+	"github.com/fossas/fossa-cli/errors"
+	"github.com/fossas/fossa-cli/exec"
+	"github.com/fossas/fossa-cli/graph"
+	"github.com/fossas/fossa-cli/pkg"
+)
+
+// Pipenv defines the interface for all pipenv tool implementation.
+type Pipenv interface {
+	Deps() (graph.Deps, error)
+}
+
+// PipenvCmd implements Pipenv by parsing command output.
+type PipenvCmd struct {
+	Dir   string
+	Graph func(dir string) (string, error)
+}
+
+// dependency is used to unmarshal the output from `pipenv graph --json-tree`
+// and store an object representing an imported dependency.
+type dependency struct {
+	Package      string `json:"package_name"`
+	Resolved     string `json:"installed_version"`
+	Target       string `json:"required_version"`
+	Dependencies []dependency
+}
+
+// New constructs a Pipenv instance that calls the pipenv build tool.
+func New(dirname string) Pipenv {
+	return PipenvCmd{
+		Dir:   dirname,
+		Graph: GraphJSON,
+	}
+}
+
+// Deps returns the dependencies of a pipenv project.
+func (p PipenvCmd) Deps() (graph.Deps, error) {
+	depGraph := graph.Deps{}
+	rawJSON, err := p.Graph(p.Dir)
+	if err != nil {
+		return depGraph, err
+	}
+
+	deps, err := getDependencies(rawJSON)
+	if err != nil {
+		return depGraph, err
+	}
+
+	depGraph.Direct = getDirectDeps(deps)
+	depGraph.Transitive = getTransitiveDeps(deps)
+	return depGraph, nil
+}
+
+func getDependencies(graphJSONFile string) ([]dependency, error) {
```

Comment:
[nit] I'm not sure `graphJSONFile` makes sense. You're passing a string, not a path to a file